src/library/export/trackexport.cpp

```diff
@@ -0,0 +1,129 @@
+#include "library/export/trackexport.h"
+
+#include <QFileInfo>
+#include <QMessageBox>
+#include <QDebug>
+
+void TrackExport::run() {
+    int i = 0;
+    for (const auto& sourceFilename : m_filenames) {
```

Comment:
actually I need to emit after i is incremented, so I'm keeping it the way I had it